Source/control.cpp

```diff
@@ -2046,21 +2046,21 @@ void control_remove_gold(int pnum, int gold_index)
 		if (plr[pnum].InvList[gi]._ivalue > 0)
 			SetGoldCurs(pnum, gi);
 		else
-			RemoveInvItem(pnum, gi);
+			inventory->RemoveItem(pnum, gi);

```

Comment:
Ok so it's not only the UI but the logic also? Then, I disagree with the way this is being handled.

The inventory is part of a player's hero, not a completely separate entity. I don't think it makes sense to have a "context-less" inventory class and then pass the player number to all calls.

Instead, this would make much more sense:
```cpp
plr[pnum].Inventory.RemoveItem(gi);
```
```

Reply:
If the UI gets its own class you can pass `plr[pnum].Inventory` to the UI as well.